[quant] update per-channel observer min/max_val attribute names

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62345

Summary:
This PR updates the attribute names from min_vals to min_val. the motivation for this is to keep the attribute name consistent with per-tensor observers so that dependencies (like FusedMovingAvgObsFakeQuantize) don't need to differentiate between the two observer types to access the attributes.

It also adds some BC tests to make sure that observers saved earlier with min_vals/max_vals can be loaded depending on the state_dict version.
Note: Scriptability of the observers isn't fully supported yet, so we aren't testing for that in this PR.

Test Plan:
python test/test_quantization.py TestSerialization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30003700](https://our.internmc.facebook.com/intern/diff/D30003700)

supriyar committed:
[quant] update per-channel observer min/max_val attribute names

Summary:
This PR updates the attribute names from min_vals to min_val. the motivation for this is to keep the attribute name consistent with per-tensor observers so that dependencies (like FusedMovingAvgObsFakeQuantize) don't need to differentiate between the two observer types to access the attributes.

It also adds some BC tests to make sure that observers saved earlier with min_vals/max_vals can be loaded depending on the state_dict version.
Note: Scriptability of the observers isn't fully supported yet, so we aren't testing for that in this PR.

Test Plan:
python test/test_quantization.py TestSerialization

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29968871).

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29968871).

vkuzo commented on torch/quantization/observer.py:
```diff
@@ -606,8 +606,9 @@ class PerChannelMinMaxObserver(_ObserverBase):
-    max_vals: torch.Tensor
+    min_val: torch.Tensor
+    max_val: torch.Tensor
+    _version = 3
```
should we move this to `_ObserverBase`, and modify that docblock?

vkuzo commented on torch/quantization/observer.py:
```diff
@@ -689,26 +689,30 @@ def _load_from_state_dict(
         for name in local_state:
             key = prefix + name
             if key in state_dict:
                 val = state_dict[key]
-                # Custom handling to allow loading min_vals or max_vals
+                # Custom handling to allow loading min_val or max_val
                 # of size N into uninitialized buffers of size 0. The
                 # buffers are resized here, and the values are copied in
                 # the default state_dict loading code of the parent.
-                if name == "min_vals":
-                    self.min_vals.resize_(val.shape)
+                if name == "min_val" or name == "min_vals":
+                    self.min_val.resize_(val.shape)
                 else:
-                    self.max_vals.resize_(val.shape)
+                    self.max_val.resize_(val.shape)
                 # For torchscript module we need to update the attributes here since we do not
                 # call the `_load_from_state_dict` function defined module.py
                 if torch.jit.is_scripting():
-                    if name == "min_vals":
-                        self.min_vals.copy_(val)
+                    if name == "min_val" or name == "min_vals":
+                        self.min_val.copy_(val)
                     else:
-                        self.max_vals.copy_(val)
+                        self.max_val.copy_(val)
```
does this implicitly handle the conversion?  Would it be better to do it explicitly, something like "if version < N, read from `min_vals` and move to `min_val`?  Otherwise it's hard to see what exactly is happening for the casual reader

supriyar replied:
I tried making it a bit more easier to read while still trying to minimize code duplication

supriyar committed:
Update on "[quant] update per-channel observer min/max_val attribute names"


Summary:
This PR updates the attribute names from min_vals to min_val. the motivation for this is to keep the attribute name consistent with per-tensor observers so that dependencies (like FusedMovingAvgObsFakeQuantize) don't need to differentiate between the two observer types to access the attributes.

It also adds some BC tests to make sure that observers saved earlier with min_vals/max_vals can be loaded depending on the state_dict version.
Note: Scriptability of the observers isn't fully supported yet, so we aren't testing for that in this PR.

Test Plan:
python test/test_quantization.py TestSerialization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29968871](https://our.internmc.facebook.com/intern/diff/D29968871)

[ghstack-poisoned]

supriyar committed:
Update on "[quant] update per-channel observer min/max_val attribute names"


Summary:
This PR updates the attribute names from min_vals to min_val. the motivation for this is to keep the attribute name consistent with per-tensor observers so that dependencies (like FusedMovingAvgObsFakeQuantize) don't need to differentiate between the two observer types to access the attributes.

It also adds some BC tests to make sure that observers saved earlier with min_vals/max_vals can be loaded depending on the state_dict version.
Note: Scriptability of the observers isn't fully supported yet, so we aren't testing for that in this PR.

Test Plan:
python test/test_quantization.py TestSerialization

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29968871](https://our.internmc.facebook.com/intern/diff/D29968871)

[ghstack-poisoned]

HDCharles reviewed (approved):
seems good

may want to consider also changing the api's to match the name schem. see e.g. line 367: reset_min_max_vals 
(not sure how to reference a line that wasn't changed)

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29968871).

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29968871).

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29968871).

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30003700).
